avm/ptn/app/container-job-toolkit/main.bicep

```
@@ -0,0 +1,341 @@
+param deployInVnet bool = false
+
+@description('Conditional. A new private DNS Zone will be created. Setting to `false` requires an existing private DNS zone `privatelink.vaultcore.azure.net`. Required if `deployInVnet` is `true`.')
+param deployDnsZoneKeyVault bool = true
+
+@description('Conditional. A new private DNS Zone will be created. Setting to `false` requires an existing private DNS zone `privatelink.azurecr.io`. Required if `deployInVnet` is `true`.')
+param deployDnsZoneContainerRegistry bool = true
+
+@description('Conditional. The address prefix for the virtual network needs to be at least a /16. Three subnets will be created (the first /24 will be used for private endpoints, the second /24 for service endpoints and the second /23 is used for the workload). Required if `zoneRedundant` for consumption plan is desired or `deployInVnet` is `true`.')
+param addressPrefix string = '10.50.0.0/16' // set a default value for the cidrSubnet calculation, even if not used
+
+import { securityRulesType } from 'modules/deploy_services.bicep'
+@description('Optional. Network security group, that will be added to the workload subnet.')
+param customNetworkSecurityGroups securityRulesType
+
+// container related parameters
+// -------------------------
+@description('Required. The container image source that will be copied to the Container Registry and used to provision the job.')
+@metadata({ example: 'mcr.microsoft.com/k8se/quickstart-jobs:latest' })
+param containerImageSource string
+
+@description('Optional. The new image name in the ACR. You can use this to import a publically available image with a custom name for later updating from e.g., your build pipeline. You should skip the registry name when specifying a custom value, as it is added automatically. If you leave this empty, the original name will be used (with the new registry name).')
+@metadata({ example: 'application/frontend:latest' })
+param newContainerImageName string?
+
+@description('Optional. The flag that indicates whether the existing image in the Container Registry should be overwritten.')
+param overwriteExistingImage bool = false
+
+@description('Optional. The cron expression that will be used to schedule the job.')
+@metadata({ default: '0 0 * * * // every day at midnight' })
+param cronExpression string = '0 0 * * *'
+
+@description('Optional. The CPU resources that will be allocated to the Container Apps Job.')
+param cpu string = '1'
+
+@description('Optional. The memory resources that will be allocated to the Container Apps Job.')
+param memory string = '2Gi'
+
+@description('Optional. The environment variables that will be added to the Container Apps Job.')
+@metadata({
+  example: '''[
+  {
+    name: 'ENV_VAR_NAME'
+    value: 'ENV_VAR_VALUE'
+  }
+  {
+    name: 'APPINSIGHTS_INSTRUMENTATIONKEY'
+    secretRef: 'applicationinsights-connection-string'
+  }
+]'''
+})
+param environmentVariables environmentVariablesType[]?
```

Please keep UDT singular
```suggestion
param environmentVariables environmentVariableType[]?
```